### PR TITLE
[fix](compile) BE compile fix on macos

### PR DESCRIPTION
**Pull request by englefly**
## Proposed changes

Issue Number: close #xxx

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by englefly** [comment]
run buildall

**Comment by englefly** [comment]
run buildall

**Comment by englefly** [comment]
run buildall

**Comment by englefly** [comment]
run buildall

**Comment by englefly** [comment]
run buildall

File: be/src/agent/cgroup_cpu_ctl.cpp [hunk 1]
```diff
@@ -120,7 +122,7 @@ Status CgroupV1CpuCtl::add_thread_to_cgroup() {
     if (!_init_succ) {
         return Status::OK();
     }
-    int tid = static_cast<int>(syscall(SYS_gettid));
+    int tid = static_cast<int>(Thread::current_thread_id());
```
Comment (wangbo) [inline be/src/agent/cgroup_cpu_ctl.cpp:125]: int tid=0;
#if defined(__linux__)
    tid = static_cast<int>(Thread::current_thread_id());
#elseif #if defined(__APPLE__) // unsupported now
   tid=0;
#endif

**Review by wangbo — APPROVED** [review]
LGTM

**Comment by englefly** [comment]
run buildall

**Review by HappenLee — APPROVED** [review]
LGTM

**Comment by englefly** [comment]
run buildall

**Review by LiBinfeng-01 — APPROVED** [review]

**Pull request merged**